gossipd/gossip_store.c

```diff
@@ -0,0 +1,48 @@
+#include "gossip_store.h"
+
+#include <ccan/endian/endian.h>
+#include <ccan/read_write_all/read_write_all.h>
+#include <common/status.h>
+#include <fcntl.h>
+
+#define GOSSIP_STORE_FILENAME "gossip_store"
+
+struct gossip_store {
+	int read_fd, write_fd;
+};
+
+struct gossip_store *gossip_store_new(const tal_t *ctx)
+{
+	struct gossip_store *gs = tal(ctx, struct gossip_store);
+	gs->write_fd = open(GOSSIP_STORE_FILENAME, O_RDWR|O_APPEND|O_CREAT, 0600);
+	gs->read_fd = open(GOSSIP_STORE_FILENAME, O_RDONLY);
```

Comment:
We `open` here, but never `close`. Granted this object has same duration as lifetime of daemon, but I think better for cleanliness.  Probably install a destructor on the `gossip_store` that closes the fd.

Reply:
Agreed, close in destructor is clenaest. But also, you don't need two fds here at all.

1. Keep an off_t for where you're up to in reading.
2. Use pread().
3. You don't need read_all or write_all, since these are normal files and read/write have those semantics.

Reply:
I'll add a destructor. But I don't see much of a difference between keeping offsets and using `pread`/`pwrite` instead of just keeping two fds, aside from saving one fd which is not really a bottleneck for `gossipd` anyway.

Good point about not needing `write_all` and `read_all` I'll use those instead.